crates/matrix-sdk/src/encryption/backups/mod.rs

```diff
@@ -387,7 +397,38 @@ impl Backups {
     /// This method will request info about the current backup from the
     /// homeserver and if a backup exits return `true`, otherwise `false`.
     pub async fn exists_on_server(&self) -> Result<bool, Error> {
-        Ok(self.get_current_version().await?.is_some())
+        let exists_on_server = self.get_current_version().await?.is_some();
+        *self.cached_exists_on_server.write().unwrap() = Some(exists_on_server);
```

Comment:
Now that I've moved the cached value down 3 layers into `BackupClientState`, this would require us to expose the internals from there in `Backups`, which feels bad to me.

I would argue that starting two requests is not a problem in this case, since that is how the method behaved before, and especially now that we've renamed the method to make it clearer that we are making a request. What do you think?